packages/purpose-process/src/routers/PurposeRouter.ts

```diff
@@ -357,7 +354,7 @@ const purposeRouter = (
     )
     .delete(
       "/purposes/:purposeId/versions/:versionId",
-      authorizationMiddleware([ADMIN_ROLE, INTERNAL_ROLE]),
```

Comment:
just for confirmation: are there are internal logics that need this endpoint? For example the revocation of delegations

Reply:
I checked, and there aren't - the delegation revocation calls a dedicated endpoint: `DELETE /internal/delegations/:delegationId/purposes/:id`

Also, calling this endpoint with an internal token would have caused a runtime error because the endpoint checks that the caller organization is the consumer or delegate, but internal tokens don't have an `organizationId` 🥲 
This is the issue of how we handle tokens and roles - we always assume we have organizationId, userId, etc. in the authdata we save in the context, but that's not true for internal/m2m/maintenance tokens. I'm proposing an improvement to solve this issue in #1611